src/System.Management.Automation/engine/runtime/Binding/Binders.cs

```diff
@@ -4258,7 +4248,7 @@ private DynamicMetaObject GetIndexMultiDimensionArray(DynamicMetaObject target,
                 }
             }
 
-            if (getterParams.Length == 1 && getterParams[0].ParameterType == typeof(int) && CanIndexFromEndWithNegativeIndex(target))
+            if (CanIndexFromEndWithNegativeIndex(target, getter, getterParams))
             {
                 // PowerShell supports negative indexing for some types (specifically, types implementing IList or IList<T>).
```

Comment:
This comment should be updated.

Reply:
Good catch, I've updated the comment to fully explain the supported scenarios.